### PR TITLE
Fix edge case with "null" predicate

### PR DESCRIPTION
**Pull request by jlatkiewicz**

### CODE REVIEW

File: yare-model-validators/src/test/java/com/sabre/oss/yare/model/validator/ReferenceValidatorTest.java [hunk 1]
```diff
@@ -65,6 +65,20 @@ void shouldFailOnMissingPredicate() {
         );
     }
 
+    @Test
+    void shouldFailOnNullPredicate() {
```
Comment (adamszadkowski) [inline yare-model-validators/src/test/java/com/sabre/oss/yare/model/validator/ReferenceValidatorTest.java]: This is duplication of existing shouldFailOnMissingPredicate test. Please remove it.

File: yare-dsl/src/main/java/com/sabre/oss/yare/dsl/RuleDsl.java [hunk 2]
```diff
@@ -674,11 +674,12 @@ private Rule createRule() {
             List<com.sabre.oss.yare.core.model.Expression.Action> actions = this.actions.stream()
                     .map(this::createAction)
                     .collect(Collectors.toList());
+            com.sabre.oss.yare.core.model.Expression predicateExpression = predicate != null ? predicate.getExpression(null, this) : null;
```
Comment (adamszadkowski) [inline yare-dsl/src/main/java/com/sabre/oss/yare/dsl/RuleDsl.java]: This change is not covered with test in RuleDslTest.
